openshift/message-board/account-micro/src/main/java/org/hibernate/demo/account/UserRepo.java

```diff
@@ -0,0 +1,42 @@
+package org.hibernate.demo.account;
+
+import java.util.List;
+import javax.enterprise.context.ApplicationScoped;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+import javax.validation.Valid;
+
+@ApplicationScoped
+public class UserRepo {
+
+	@PersistenceContext
+	private EntityManager em;
+
+	public UserRepo() {
+	}
+
+	public UserRepo(EntityManager em) {
+		this.em = em;
+	}
+
+	public void add(@Valid User user) {
+		em.persist( user );
+	}
+
+	public User findByUserName(String userName) {
+		Query query = em.createQuery( "from User u where u.userName = :userName" );
+		query.setParameter( "userName", userName );
+		return (User) query.getSingleResult();
+	}
+
+	public void delete(String userName) {
+		Query query = em.createQuery( "delete from User u where u.userName = :userName" );
+		query.setParameter( "userName", userName );
+		query.executeUpdate();
```

Comment:
I don't think this is 2nd level cache friendly. I would do a load then a delete.

That makes sense as I think, in a real application, you should probably return an error if you're trying to delete a user who does not exist.